src/main/java/trackr/model/task/TaskDeadline.java

```diff
@@ -39,15 +40,28 @@ public TaskDeadline(String deadline) {
      * the parsed date is today's date or after today's date.
      */
     public static boolean isValidTaskDeadline(String test) {
-        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        }
+
         try {
+            DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd/MM/yyyy");
             LocalDate localDateTest = LocalDate.parse(test, dtf);
```

Comment:
is there a way to check if the given string deadline is a valid date in the calendar without having to parse it another time?